DOC-3406: Fixed links in CE 4.5 - Developing CorDapps section (#263)
corda/corda-docs-portal: content/en/platform/corda/4.5/enterprise/cordapps/cordapp-compatibility.md

In order to develop a CorDapp for compatibility between Corda and Corda Enterprise follow these steps:

> * Ensure your CorDapp is designed per [Structuring a CorDapp](writing-a-cordapp.md) and annotated according to [CorDapp separation](../../../../../../../en/platform/corda/4.5/enterprise/cordapps/cordapp-build-systems.html#cordapp-separation-ref).

Reviewer: ```suggestion
* Ensure your CorDapp is designed per [CorDapp structure and format](writing-a-cordapp.md) and annotated according to [CorDapp separation](../../../../../../../en/platform/corda/4.5/enterprise/cordapps/cordapp-build-systems.html#separation-of-cordapp-contracts-flows-and-services).
```

In particular, it is critical to separate the consensus-critical parts of your application (contracts, states and their dependencies) from
the rest of the business logic (flows, APIs, etc).
The former - the **CorDapp kernel** - is the Jar that will be attached to transactions creating/consuming your states and is the Jar
that any node on the network verifying the transaction must execute.

{{< note >}}
It is also important to understand how to manage any dependencies a CorDapp may have on 3rd party libraries and other CorDapps.

> Please read [Setting your dependencies](../../../../../../../en/platform/corda/4.5/enterprise/cordapps/cordapp-build-systems.html#cordapp-dependencies-ref) to understand the options and recommendations with regards to correctly Jar’ing CorDapp dependencies.

Reviewer: ```suggestion
Please read [Setting your dependencies](../../../../../../../en/platform/corda/4.5/enterprise/cordapps/cordapp-build-systems.html#setting-your-dependencies) to understand the options and recommendations with regards to correctly Jar’ing CorDapp dependencies.
```

{{< /note >}}

* Compile this **CorDapp kernel** Jar once, and then depend on it from your workflows Jar. In terms of Corda depdendencies,this should only
depend on the `corda-core` package from the Corda Open Source distribution.

{{< note >}}

> As of Corda 4 it is recommended to use [CorDapp Jar signing](../../../../../../../en/platform/corda/4.5/enterprise/cordapps/cordapp-build-systems.html#cordapp-build-system-signing-cordapp-jar-ref) to leverage the new signature constraints functionality.

Reviewer: ```suggestion
As of Corda 4 it is recommended to use [CorDapp Jar signing](../../../../../../../en/platform/corda/4.5/enterprise/cordapps/cordapp-build-systems.html#signing-the-cordapp-jar) to leverage the new signature constraints functionality.
```
